freqtrade/data/entryexitanalysis.py

```diff
@@ -0,0 +1,259 @@
+                       'profit_abs_median', 'profit_abs_mean', 'median_profit_pct',
+                       'mean_profit_pct', 'total_profit_pct']
+        sortcols = ['profit_abs_sum', 'enter_reason']
+
+        new['median_profit_pct'] = new['median_profit_pct'] * 100
+        new['mean_profit_pct'] = new['mean_profit_pct'] * 100
+        new['total_profit_pct'] = new['total_profit_pct'] * 100
+
+        _print_table(new, sortcols)
+
+
+def _print_results(analysed_trades, stratname, group,
+                   enter_reason_list, exit_reason_list,
+                   indicator_list, columns=None):
+
+    if columns is None:
+        columns = ['pair', 'open_date', 'close_date', 'profit_abs', 'enter_reason', 'exit_reason']
+
+    bigdf = pd.DataFrame()
+    for pair, trades in analysed_trades[stratname].items():
+        bigdf = pd.concat([bigdf, trades], ignore_index=True)
+
+    if bigdf.shape[0] > 0 and ('enter_reason' in bigdf.columns):
+        if group is not None:
+            glist = group.split(",")
+            _do_group_table_output(bigdf, glist)
+
+        if enter_reason_list is not None and not enter_reason_list == "all":
+            enter_reason_list = enter_reason_list.split(",")
+            bigdf = bigdf.loc[(bigdf['enter_reason'].isin(enter_reason_list))]
+
+        if exit_reason_list is not None and not exit_reason_list == "all":
+            exit_reason_list = exit_reason_list.split(",")
+            bigdf = bigdf.loc[(bigdf['exit_reason'].isin(exit_reason_list))]
+
+        if indicator_list is not None:
+            if indicator_list == "all":
+                print(bigdf)
+            else:
+                available_inds = []
+                for ind in indicator_list.split(","):
+                    if ind in bigdf:
+                        available_inds.append(ind)
+                ilist = ["pair", "enter_reason", "exit_reason"] + available_inds
+                _print_table(bigdf[ilist], sortcols=['exit_reason'], show_index=False)
+        else:
+            _print_table(bigdf[columns], sortcols=['pair'], show_index=False)
+    else:
+        print("\\_ No trades to show")
+
+
+def _print_table(df, sortcols=None, show_index=False):
+    if (sortcols is not None):
+        data = df.sort_values(sortcols)
+    else:
+        data = df
+
+    print(
+        tabulate(
+            data,
+            headers='keys',
+            tablefmt='psql',
+            showindex=show_index
+        )
+    )
+
+
+def process_entry_exit_reasons(backtest_dir: Path,
+                               pairlist: List[str],
+                               strategy_name: str,
+                               analysis_groups: Optional[str] = "0,1,2",
+                               enter_reason_list: Optional[str] = "all",
+                               exit_reason_list: Optional[str] = "all",
+                               indicator_list: Optional[str] = None):
+
+    try:
+        trades = load_backtest_data(backtest_dir, strategy_name)
```

Comment:
Consider using load_backtest_stats here (first) - that way you can derive the strategy.

If we assume users MUST insert a strategy for this command explicitly - we should check for that.

Reply:
Done